Source/BugsnagEvent.m

```diff
@@ -664,14 +667,14 @@ - (void)setUser:(NSString *_Nullable)userId
  * @param metadata the KSCrash report metadata
  * @return the user, or nil if not available
  */
-- (BugsnagUser *)parseUser:(NSDictionary *)metadata {
+- (BugsnagUser *)parseUser:(NSDictionary *)metadata deviceAppHash:(NSString *)deviceAppHash {
     NSMutableDictionary *user = [metadata[BSGKeyUser] mutableCopy];
     if (user == nil) {
         user = [NSMutableDictionary dictionary];
     }
 
-    if (!user[BSGKeyId] && self.device.id) { // if device id is null, don't set user id to default
-        BSGDictSetSafeObject(user, self.deviceAppHash, BSGKeyId);
+    if (!user[BSGKeyId]) {
```

Comment:
Is deviceAppHash now guaranteed non-null?

Reply:
`deviceAppHash` should be non-null - I think it'd be preferable to alter this to `BSGDictInsertIfNotNil`, and also pass in `self.device.id` if it's available to retain the old logic.